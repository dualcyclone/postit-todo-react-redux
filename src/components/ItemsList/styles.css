.itemsList-ul {
  list-style: none;
}

.itemsList-ul {
  overflow: hidden;
  padding: 3em;
}
