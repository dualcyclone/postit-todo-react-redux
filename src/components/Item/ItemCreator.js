import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { addItem } from '../../main/store/actions'
import Item from './Item'
import './styles.css'

export class ItemCreator extends Component {
  constructor() {
    super()

    this.state = {
      inputValue: ''
    }

    this.inputField = React.createRef()
  }

  handleAddItem() {
    const { inputValue } = this.state

    if (inputValue) {
      this.props.onAdd(inputValue)
    }

    this.setState({ inputValue: '' })
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value
    })
  }

  render() {
    return (
      <Item ElementType="div" className="itemCreator">
        <p>Add another item...</p>
        <textarea
          ref={this.inputField}
          className="itemCreator-input"
          type="text"
          rows="8"
          placeholder="What do you need to do?"
          onKeyPress={(e) => {
            if (e.key === 'Enter') {
              e.preventDefault()
              this.handleAddItem()
            }
          }}
          value={this.state.inputValue}
          onChange={evt => this.updateInputValue(evt)}
        />
        <button onClick={() => this.handleAddItem()} className="icon icono-plusCircle">&nbsp</button>
      </Item>
    )
  }
}

ItemCreator.propTypes = {
  onAdd: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => ({
  onAdd: newItem => dispatch(addItem(newItem))
})

export default connect(
  null,
  mapDispatchToProps
)(ItemCreator)
