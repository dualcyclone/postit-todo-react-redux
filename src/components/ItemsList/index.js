import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Item from '../Item';
import './styles.css';

const ItemsList = ({ items }) => {
  return (
    <div>
      { items.length === 0 &&
        <p id={'items-missing'}>
          Add some tasks above.
        </p>
      }
      { items.length > 0 &&
        <ul className={'itemsList-ul'}>
          { items.map(item => (
            <Item key={item.id} item={item}>{ item.content }</Item>
          )) }
        </ul>
      }
    </div>
  );
}

ItemsList.propTypes = {
  items: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  return {
    items: state.items
  };
};

export default connect(
  mapStateToProps
)(ItemsList);
