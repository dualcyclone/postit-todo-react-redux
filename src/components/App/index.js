import React from 'react';
import ItemCreator from '../Item/ItemCreator';
import ItemsList from '../ItemsList';
import './styles.css';

const App = () => (
  <div className="app">
    <div>
      <ItemCreator />
      <ItemsList />
    </div>
  </div>
);

export default App;
