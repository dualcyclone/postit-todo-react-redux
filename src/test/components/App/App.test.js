import React from 'react'
import { expect } from 'chai'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { App } from '../../../components/App/index'
import ItemCreator from '../../../components/Item/ItemCreator'
import ItemsList from '../../../components/ItemsList'

const jestExpect = global.expect

configure({adapter: new Adapter()})

describe('App', () => {
  it('Should render without failing', () => {
    const renderedItem = shallow(<App />)

    jestExpect(toJson(renderedItem)).toMatchSnapshot()
  })

  it('Should render with the class of "app"', () => {
    const renderedItem = shallow(
      <App />
    )

    expect(renderedItem.is('div')).to.equal(true)
    expect(renderedItem.is('.app')).to.equal(true)
  })

  describe('Should contain the following children', () => {
    let renderedItem

    beforeEach(() => {
      renderedItem = shallow(
        <App />
      )
    })

    it('Contains two children', () => {
      expect(renderedItem.find('div').children().length).to.equal(2)
    })

    it('ItemCreator exists', () => {
      expect(renderedItem.find('div').find(ItemCreator)).not.to.be.undefined
    })

    it('ItemsList exists', () => {
      expect(renderedItem.find('div').find(ItemsList)).not.to.be.undefined
    })
  })
})
