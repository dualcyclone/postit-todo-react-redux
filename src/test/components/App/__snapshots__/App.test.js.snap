// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render without failing 1`] = `
<div
  className="app"
>
  <Connect(ItemCreator) />
  <Connect(ItemsList) />
</div>
`;
